hbase-server/src/main/java/org/apache/hadoop/hbase/tool/CanaryTool.java

```diff
@@ -122,6 +125,35 @@
  */
 @InterfaceAudience.LimitedPrivate(HBaseInterfaceAudience.TOOLS)
 public class CanaryTool implements Tool, Canary {
+  public static final String HBASE_CANARY_INFO_PORT = "hbase.canary.info.port";
+
+  public static final int DEFAULT_CANARY_INFOPORT = 16050;
+
+  public static final String HBASE_CANARY_INFO_BINDADDRESS = "hbase.canary.info.bindAddress";
+
+
+  private void putUpWebUI() throws IOException {
+    if (zookeeperMode) {
+      LOG.info("WebUI is not supported in Zookeeper mode");
```

Comment:
nit: could skip these warnings (and the whole function body) when `HBASE_CANARY_INFO_PORT == -1`

Reply:
> nit: could skip these warnings (and the whole function body) when `HBASE_CANARY_INFO_PORT == -1`

will deal with it later